src/action-registry/dto/update-action.dto.ts

```diff
@@ -0,0 +1,7 @@
+import { IsNotEmpty, IsString } from 'class-validator';
+
+export class UpdateActionDto {
+  @IsString()
+  @IsNotEmpty()
+  action: string;
+}
```

Comment:
Please map existing types instead of duplicating
```suggestion
import { PickType } from '@nestjs/swagger';
import { CreateActionDto } from './create-action.dto';

export class UpdateActionDto extends PickType(CreateActionDto, ['action'] as const) {}
```